bindings/node/test/clientEncryption.test.js

```diff
@@ -0,0 +1,139 @@
+  });
+
+  [
+    {
+      name: 'local',
+      kmsProviders: { local: { key: Buffer.alloc(96) } }
+    },
+    {
+      name: 'aws',
+      kmsProviders: { aws: { accessKeyId: 'example', secretAccessKey: 'example' } },
+      options: { masterKey: { region: 'region', key: 'cmk' } }
+    }
+  ].forEach(providerTest => {
+    it(`should create a data key with the "${providerTest.name}" KMS provider`, function(done) {
+      const providerName = providerTest.name;
+      const encryption = new ClientEncryption(client, {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: providerTest.kmsProviders
+      });
+
+      const dataKeyOptions = providerTest.options || {};
+      encryption.createDataKey(providerName, dataKeyOptions, (err, dataKey) => {
+        expect(err).to.not.exist;
+        expect(dataKey._bsontype).to.equal('Binary');
+
+        client
+          .db('client')
+          .collection('encryption')
+          .findOne({ _id: dataKey }, (err, doc) => {
+            expect(err).to.not.exist;
+            expect(doc).to.have.property('masterKey');
+            expect(doc.masterKey).to.have.property('provider');
+            expect(doc.masterKey.provider).to.eql(providerName);
+            done();
+          });
+      });
+    });
+  });
+
+  it('should explicitly encrypt and decrypt with the "local" KMS provider', function(done) {
+    const encryption = new ClientEncryption(client, {
+      keyVaultNamespace: 'client.encryption',
+      kmsProviders: { local: { key: Buffer.alloc(96) } }
+    });
+
+    encryption.createDataKey('local', (err, dataKey) => {
+      expect(err).to.not.exist;
+
+      const encryptOptions = {
+        keyId: dataKey,
+        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+      };
+
+      encryption.encrypt('hello', encryptOptions, (err, encrypted) => {
+        expect(err).to.not.exist;
+        expect(encrypted._bsontype).to.equal('Binary');
+        expect(encrypted.sub_type).to.equal(6);
+
+        encryption.decrypt(encrypted, (err, decrypted) => {
+          expect(err).to.not.exist;
+          expect(decrypted).to.equal('hello');
+          done();
+        });
+      });
+    });
+  });
+
+  it.skip('should explicitly encrypt and decrypt with the "aws" KMS provider', function(done) {
+    const encryption = new ClientEncryption(client, {
+      keyVaultNamespace: 'client.encryption',
+      kmsProviders: { aws: { accessKeyId: 'example', secretAccessKey: 'example' } }
+    });
+
+    const dataKeyOptions = {
+      masterKey: { region: 'region', key: 'cmk' }
+    };
+
+    encryption.createDataKey('aws', dataKeyOptions, (err, dataKey) => {
+      expect(err).to.not.exist;
+
+      const encryptOptions = {
+        keyId: dataKey,
+        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+      };
+
+      encryption.encrypt('hello', encryptOptions, (err, encrypted) => {
+        expect(err).to.not.exist;
+        expect(encrypted).to.have.property('v');
```

Comment:
What is the v property?

Reply:
The `v` is a requirement from embedding libmongocrypt. In order to encrypt a single value we need to ensure that it's wrapped in a document because BSON only supports documents. By wrapping in `v`, we ensure we can handle documents and scalar values through encryption